v4/export/sql.go

```diff
@@ -401,6 +397,49 @@ func CheckTiDBWithTiKV(db *sql.DB) (bool, error) {
 	return count > 0, nil
 }
 
+func getSnapshot(db *sql.DB) (string, error) {
+	str, err := ShowMasterStatus(db, showMasterStatusFieldNum)
+	if err != nil {
+		return "", err
+	}
+	return str[snapshotFieldIndex], nil
+}
+
+func isUnknownSystemVariableErr(err error) bool {
+	return strings.Contains(err.Error(), "Unknown system variable")
+}
+
+func resetDBWithSessionParams(db *sql.DB, dsn string, params map[string]string) (*sql.DB, error) {
+	support := make(map[string]string)
+	for k, v := range params {
+		v = wrapQuotes(v)
+		s := fmt.Sprintf("SET SESSION %s = %s", k, v)
+		_, err := db.Exec(s)
```

Comment:
```suggestion
		s := fmt.Sprintf("SET SESSION %s = ?", k)
		_, err := db.Exec(s, v)
```

Reply:
I use this format at first, but at last this failed when I try to set `tidb_mem_quota_query`.
```log
[2020/06/30 18:39:05.433 +08:00] [ERROR] [main.go:191] ["dump failed error stack info"] [error="Error 1231: Variable 'tidb_mem_quota_query' can't be set to the value of '%!s(MISSING)'"] [errorVerbose="err = Error 1231: Variable 'tidb_mem_quota_query' can't be set to the value of '%!s(MISSING)'\ngoroutine 1 [running]:\nruntime/debug.Stack(0x19fc520, 0xc0001fc220, 0x19fd9e0)\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0x9d\ngithub.com/pingcap/dumpling/v4/export.withStack(0x19fc520, 0xc0001fc220, 0xc000038000, 0xc000152570)\n\t/Users/chauncy/code/goPath/src/github.com/pingcap/dumpling/v4/export/error.go:59 +0x8d\ngithub.com/pingcap/dumpling/v4/export.resetDBWithSessionParams(0xc0001fe000, 0xc000152540, 0x2a, 0xc0001d2600, 0x2a, 0x0, 0x0)\n\t/Users/chauncy/code/goPath/src/github.com/pingcap/dumpling/v4/export/sql.go:389 +0x684\ngithub.com/pingcap/dumpling/v4/export.Dump(0x1a13400, 0xc000038090, 0xc00012cea0, 0x0, 0x0)\n\t/Users/chauncy/code/goPath/src/github.com/pingcap/dumpling/v4/export/dump.go:101 +0x327\nmain.main()\n\t/Users/chauncy/code/goPath/src/github.com/pingcap/dumpling/cmd/dumpling/main.go:189 +0x132f\n"] [stack="github.com/pingcap/log.Error\n\t/Users/chauncy/code/goPath/pkg/mod/github.com/pingcap/log@v0.0.0-20200511115504-543df19646ad/global.go:42\nmain.main\n\t/Users/chauncy/code/goPath/src/github.com/pingcap/dumpling/cmd/dumpling/main.go:191\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:203"]
```

Reply:
arguments:
```log
Running command: ['/Users/chauncy/code/goPath/src/github.com/pingcap/dumpling/bin/dumpling', '-T', 'TICASE_3204_WTHYZC.dumpsnap', '--dump-empty-database=false', '--snapshot', '2020-06-30 18:39:03', '-h', '127.0.0.1', '-P', '4000', '-u', 'root', '-p', '', '-o', '/var/folders/sr/_hc37vmj0db6ht4_w6rw34880000gn/T/tmptv4miiea']
```

Reply:
@lichunzhu why is it `%!s(MISSING)`

Reply:
Before db.Exec this variable's values is `"34359738368"`.  Maybe after translation it becomes `%!s(MISSING)`.